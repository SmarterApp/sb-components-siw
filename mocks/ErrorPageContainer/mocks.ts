import { pageType, ErrorPageContainerProps } from "@src/index";

export const serverErrMock: ErrorPageContainerProps = {
  errorCode: pageType.ServerError,
  errorMsg: "you did a bad thing that caused errors",
  description: "stop doing that thing that caused errors"
};

export const notFoundErrMock: ErrorPageContainerProps = {
  errorCode: pageType.NotFound,
  errorMsg: "This item cannot be found",
  description: "Please try another search"
};
