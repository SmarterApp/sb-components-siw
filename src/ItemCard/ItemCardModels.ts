import { GradeLevels } from "../GradeLevels/GradeLevels";

export interface ItemCardModel {
  selected?: boolean;
  selectionIndex?: number;
  bankKey: number;
  itemKey: number;
  stimulusKey?: number;
  itemDifficulty: string;
  title: string;
  grade: GradeLevels;
  gradeLabel: string;
  subjectCode: string;
  subjectLabel: string;
  testName?: string;
  testOrder?: number;
  testNameInPrintCart?: string;
  testOrderInPrintCart?: number;
  claimCode: string;
  claimLabel: string;
  targetHash: number;
  targetShortName: string;
  targetId: string;
  targetDescription: string;
  interactionTypeCode: string;
  interactionTypeLabel: string;
  isPerformanceItem: boolean;
  brailleOnlyItem?: boolean;
  domain?: string;
  depthOfKnowledge?: string;
  commonCoreStandardId: string;
  ccssDescription?: string;
  calculator?: boolean;
  releaseDate?: string;
  selectedBrailleTypes?: string[];
  availableBrailleTypes: BrailleTypeModel[];
}

export function itemIdEqual(a: ItemCardModel, b: ItemCardModel) {
  return a.itemKey === b.itemKey && a.bankKey === b.bankKey;
}

export interface BrailleTypeModel {
  selectionCode: string;
  label: string;
  disabled: boolean;
  hidden: boolean;
}
