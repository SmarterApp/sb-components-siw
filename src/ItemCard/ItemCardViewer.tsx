import * as React from "react";
import {
  ItemViewerFrame,
  RubricTable,
  AboutThisItemDetail,
  ItemTabs,
  Tabs,
  AboutItemModel
} from "@src/index";

export interface ItemCardViewerProps {
  item?: AboutItemModel;
}

export interface State {
  selectedTab: Tabs;
}

export class ItemCardViewer extends React.Component<
  ItemCardViewerProps,
  State
> {
  constructor(props: ItemCardViewerProps) {
    super(props);
    this.state = {
      selectedTab: "viewer"
    };
  }

  onTabChange(tab: Tabs) {
    this.setState({
      selectedTab: tab
    });
  }

  renderViewer(url: string) {
    return (
      <div className="item-content">
        <ItemViewerFrame url={url} />
      </div>
    );
  }

  renderRubric() {
    const aboutItem = this.props.item;

    if (aboutItem && aboutItem.sampleItemScoring) {
      if (aboutItem.sampleItemScoring.rubrics) {
        return (
          <div className="item-content">
            <RubricTable rubrics={aboutItem.sampleItemScoring.rubrics} />
          </div>
        );
      }
    }
  }

  renderInformation() {
    if (this.props.item) {
      const aboutItem = this.props.item;

      return (
        <div className="item-content">
          <div>
            <AboutThisItemDetail {...aboutItem} />
          </div>
        </div>
      );
    }
  }

  renderChosen() {
    if (this.props.item) {
      const selectedTab = this.state.selectedTab;
      const itemCard = this.props.item.itemCardViewModel;

      let resultElement: JSX.Element[] | JSX.Element | undefined;
      if (selectedTab === "viewer") {
        const url = `http://ivs.smarterbalanced.org/items?ids=${
          itemCard.bankKey
        }-${itemCard.itemKey}`;
        resultElement = <div> {this.renderViewer(url)} </div>;
      } else if (selectedTab === "rubric") {
        resultElement = <div> {this.renderRubric()} </div>;
      } else if (selectedTab === "information") {
        resultElement = <div> {this.renderInformation()}</div>;
      }

      return resultElement;
    }
  }
  render() {
    let rubricVisibility = false;
    if (
      this.props.item &&
      this.props.item.sampleItemScoring &&
      this.props.item.sampleItemScoring.rubrics
    ) {
      rubricVisibility = true;
    }

    return (
      <div className="item-card">
        <ItemTabs
          showRubricTab={rubricVisibility}
          changedTab={tab => this.onTabChange(tab)}
          selectedTab={this.state.selectedTab}
        />
        {this.renderChosen()}
      </div>
    );
  }
}
