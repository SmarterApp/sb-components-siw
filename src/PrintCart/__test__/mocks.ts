import { ItemCardModel } from "@src/index";
import { currentId } from "async_hooks";

export const itemHandler = jest.fn((item: ItemCardModel) => {
  return undefined;
});

export const getSelectedItemCount = jest.fn(() => {
  return;
});

export const countNumberOfItemsAfterSelection = jest.fn(
  (currentItems: ItemCardModel[], selectedItemsCount: number) => {
    return;
  }
);

export const onAddOrRemoveSelectedItems = jest.fn((item: ItemCardModel) => {
  return;
});

export const onItemsReorder = jest.fn((i: number, j: number) => {
  return;
});
