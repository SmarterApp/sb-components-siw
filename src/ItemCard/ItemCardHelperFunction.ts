import { ItemCardModel } from "./ItemCardModels";

export function getContentStandardCode(
  subjectCode: any,
  claimCode: any,
  commonCoreStandardId: any,
  ccssDescription: any
) {
  commonCoreStandardId =
    commonCoreStandardId == null || commonCoreStandardId == undefined
      ? "Not Available"
      : commonCoreStandardId;
  ccssDescription =
    ccssDescription == null || ccssDescription == undefined
      ? "Content Standard information is currently unavailable for this item."
      : ccssDescription;

  //For math items and claim from 2 to 4
  if (
    subjectCode === "MATH" &&
    (claimCode == "MATH2" || claimCode == "MATH3" || claimCode == "MATH4")
  ) {
    commonCoreStandardId = "Math Practice";
    ccssDescription =
      "Items in this claim primarily measure the Standards for Mathematical Practice rather than Content Standards.";
  }

  return { commonCoreStandardId, ccssDescription };
}

/**
 * Get number items that will go to print carts once user selects a item or multiple items
 */
export function countNumberOfItemsAfterSelection(
  items: ItemCardModel[],
  currentCount: number,
  totalAssociatedItems: any
) {
  let itemsToExcludeCounting: number[] = [];
  let count = 0;
  items.forEach(item => {
    if (!item.isPerformanceItem) {
      count = count + 1;
    } else {
      if (item.itemKey in totalAssociatedItems) {
        if (itemsToExcludeCounting.indexOf(item.itemKey) === -1) {
          itemsToExcludeCounting.push(...totalAssociatedItems[item.itemKey]);
          const associatedItemsCount =
            totalAssociatedItems[item.itemKey].length;
          count = count + associatedItemsCount;
        }
      }
    }
  });
  return count + currentCount;
}
