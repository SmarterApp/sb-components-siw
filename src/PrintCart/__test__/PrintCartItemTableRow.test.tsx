import "jsdom-global/register";
import * as React from "react";
import * as TestUtils from "react-dom/test-utils";
import { shallow, mount, render } from "enzyme";
import { itemCardList, PTassociatedItems } from "@mocks/ItemCard/mocks";

import { onAddOrRemoveSelectedItems, onItemsReorder } from "./mocks";
import {
  RubricModel,
  headerColumns,
  PrintCartItemTableRow,
  PrintCartItemTableRowProps
} from "@src/index";

describe("ItemTableRow", () => {
  const rubrics: RubricModel[] = [];
  const selectedItem = itemCardList[1];

  const props: PrintCartItemTableRowProps = {
    ItemCard: selectedItem,
    isInterimSite: true,
    TotalItemsCard: itemCardList,
    associatedItemsInPrintCart: PTassociatedItems[3206],
    onAddOrRemoveSelectedItems: onAddOrRemoveSelectedItems,
    onItemsReorder: onItemsReorder,
    columns: headerColumns,
    isLinkTable: false,
    itemSequence: 2,
    index: 1,
    testCodeToLabelMap: { "Test-Name-1": "Test 1", "Test-Name-2": "Test 2" },
    itemIdToTestNameMap: {
      3206: {
        testName: "Test 1",
        testOrder: 1
      },
      3163: {
        testName: "Test 1",
        testOrder: 2
      },
      3615: {
        testName: "Test 1",
        testOrder: 3
      },
      2928: {
        testName: "Test 2",
        testOrder: 1
      }
    }
  };

  const wrapper = mount(
    <table>
      <tbody>
        <PrintCartItemTableRow {...props} />
      </tbody>
    </table>
  );

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
