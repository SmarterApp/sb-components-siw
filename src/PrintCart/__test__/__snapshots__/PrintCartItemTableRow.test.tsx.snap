// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTableRow matches snapshot 1`] = `
<table>
  <tbody>
    <PrintCartItemTableRow
      ItemCard={
        Object {
          "availableBrailleTypes": Array [
            Object {
              "disabled": false,
              "hidden": false,
              "label": "TDS BT ECN",
              "selectionCode": "TDS_BT_ECN",
            },
            Object {
              "disabled": false,
              "hidden": false,
              "label": "DS BT UCB Label",
              "selectionCode": "TDS_BT_UCB",
            },
            Object {
              "disabled": false,
              "hidden": false,
              "label": "TDS BT ECL label",
              "selectionCode": "TDS_BT_ECL",
            },
            Object {
              "disabled": false,
              "hidden": false,
              "label": "TTDS BT EXN Label",
              "selectionCode": "TTDS_BT_EXN",
            },
          ],
          "bankKey": 187,
          "brailleOnlyItem": false,
          "calculator": false,
          "claimCode": "ELA1",
          "claimLabel": "1.Reading",
          "commonCoreStandardId": "3.R.2a",
          "grade": 1,
          "gradeLabel": "Grade 3",
          "interactionTypeCode": "MC",
          "interactionTypeLabel": "Multiple Choice",
          "isPerformanceItem": true,
          "itemDifficulty": "Easy",
          "itemKey": 3163,
          "stimulusKey": 58679,
          "subjectCode": "ELA",
          "subjectLabel": "ELA/literacy",
          "targetDescription": "Target B description",
          "targetHash": 1039,
          "targetId": "B",
          "targetShortName": "Key Details",
          "testName": "Test-Name-1",
          "testOrder": 2,
          "title": "ELA/ literacy Grade 3 Claim 1",
        }
      }
      TotalItemsCard={
        Array [
          Object {
            "availableBrailleTypes": Array [
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECN",
                "selectionCode": "TDS_BT_ECN",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "DS BT UCB Label",
                "selectionCode": "TDS_BT_UCB",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECL label",
                "selectionCode": "TDS_BT_ECL",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TTDS BT EXN Label",
                "selectionCode": "TTDS_BT_EXN",
              },
            ],
            "bankKey": 187,
            "brailleOnlyItem": false,
            "calculator": true,
            "claimCode": "MATH1",
            "claimLabel": "1.Concepts and Procedures and Logical",
            "commonCoreStandardId": "6.M.1a",
            "grade": 8,
            "gradeLabel": "Grade 6",
            "interactionTypeCode": "EQ",
            "interactionTypeLabel": "Equation Multiple Choice choice",
            "isPerformanceItem": true,
            "itemDifficulty": "Moderate",
            "itemKey": 3206,
            "stimulusKey": 56900,
            "subjectCode": "MATH",
            "subjectLabel": "Math",
            "targetDescription": "Target A description with extra text for testing",
            "targetHash": 100,
            "targetId": "A",
            "targetShortName": "",
            "testName": "Test-Name-1",
            "testOrder": 1,
            "title": "Math Grade 6 Claim 1",
          },
          Object {
            "availableBrailleTypes": Array [
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECN",
                "selectionCode": "TDS_BT_ECN",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "DS BT UCB Label",
                "selectionCode": "TDS_BT_UCB",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECL label",
                "selectionCode": "TDS_BT_ECL",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TTDS BT EXN Label",
                "selectionCode": "TTDS_BT_EXN",
              },
            ],
            "bankKey": 187,
            "brailleOnlyItem": false,
            "calculator": false,
            "claimCode": "ELA1",
            "claimLabel": "1.Reading",
            "commonCoreStandardId": "3.R.2a",
            "grade": 1,
            "gradeLabel": "Grade 3",
            "interactionTypeCode": "MC",
            "interactionTypeLabel": "Multiple Choice",
            "isPerformanceItem": true,
            "itemDifficulty": "Easy",
            "itemKey": 3163,
            "stimulusKey": 58679,
            "subjectCode": "ELA",
            "subjectLabel": "ELA/literacy",
            "targetDescription": "Target B description",
            "targetHash": 1039,
            "targetId": "B",
            "targetShortName": "Key Details",
            "testName": "Test-Name-1",
            "testOrder": 2,
            "title": "ELA/ literacy Grade 3 Claim 1",
          },
          Object {
            "availableBrailleTypes": Array [
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECN",
                "selectionCode": "TDS_BT_ECN",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "DS BT UCB Label",
                "selectionCode": "TDS_BT_UCB",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECL label",
                "selectionCode": "TDS_BT_ECL",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TTDS BT EXN Label",
                "selectionCode": "TTDS_BT_EXN",
              },
            ],
            "bankKey": 187,
            "brailleOnlyItem": false,
            "claimCode": "MATH4",
            "claimLabel": "3.Modeling/Data Analysis",
            "commonCoreStandardId": "6.M.2b",
            "grade": 8,
            "gradeLabel": "Grade 6",
            "interactionTypeCode": "MS",
            "interactionTypeLabel": "Multi Select",
            "isPerformanceItem": true,
            "itemDifficulty": "",
            "itemKey": 3615,
            "stimulusKey": 23461,
            "subjectCode": "MATH",
            "subjectLabel": "Math",
            "targetDescription": "target C description",
            "targetHash": 0,
            "targetId": "C",
            "targetShortName": "",
            "testName": "Test-Name-1",
            "testOrder": 3,
            "title": "Math Grade 6 Claim 4",
          },
          Object {
            "availableBrailleTypes": Array [
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECN",
                "selectionCode": "TDS_BT_ECN",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "DS BT UCB Label",
                "selectionCode": "TDS_BT_UCB",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TDS BT ECL label",
                "selectionCode": "TDS_BT_ECL",
              },
              Object {
                "disabled": false,
                "hidden": false,
                "label": "TTDS BT EXN Label",
                "selectionCode": "TTDS_BT_EXN",
              },
            ],
            "bankKey": 187,
            "brailleOnlyItem": false,
            "claimCode": "ELA4",
            "claimLabel": "4.Research/Inquiry",
            "commonCoreStandardId": "4.E.4a",
            "grade": 2,
            "gradeLabel": "Grade 4",
            "interactionTypeCode": "MS",
            "interactionTypeLabel": "Multi Select",
            "isPerformanceItem": false,
            "itemDifficulty": "Hard",
            "itemKey": 2928,
            "stimulusKey": 87679,
            "subjectCode": "ELA",
            "subjectLabel": "ELA/literacy",
            "targetDescription": "target D description",
            "targetHash": 2832,
            "targetId": "D",
            "targetShortName": "Evaluate Information/ sources",
            "testName": "",
            "testOrder": 2,
            "title": "ELA/ literacy Grade 4 Claim 4",
          },
        ]
      }
      associatedItemsInPrintCart={
        Array [
          Array [
            Object {
              "bankKey": 187,
              "brailleOnlyItem": false,
              "calculator": false,
              "claimCode": "ELA1",
              "claimLabel": "Reading",
              "grade": 1,
              "gradeLabel": "Grade 3",
              "interactionTypeCode": "MC",
              "interactionTypeLabel": "Multiple Choice",
              "isPerformanceItem": true,
              "itemKey": 3163,
              "subjectCode": "ELA",
              "subjectLabel": "ELA/literacy",
              "targetDescription": "Target B description",
              "targetHash": 1039,
              "targetId": "B",
              "targetShortName": "Key Details",
              "title": "ELA/ literacy Grade 3 Claim 1",
            },
          ],
          Array [
            Object {
              "bankKey": 187,
              "brailleOnlyItem": false,
              "calculator": true,
              "claimCode": "MATH1",
              "claimLabel": "Concepts and Procedures and Logical",
              "grade": 8,
              "gradeLabel": "Grade 6",
              "interactionTypeCode": "EQ",
              "interactionTypeLabel": "Equation Multiple Choice choice",
              "isPerformanceItem": true,
              "itemKey": 3206,
              "subjectCode": "MATH",
              "subjectLabel": "Math",
              "targetDescription": "Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia",
              "targetHash": 100,
              "targetId": "A",
              "targetShortName": "",
              "title": "Math Grade 6 Claim 1",
            },
          ],
          Array [
            Object {
              "bankKey": 187,
              "brailleOnlyItem": false,
              "claimCode": "MATH4",
              "claimLabel": "Modeling/Data Analysis",
              "grade": 8,
              "gradeLabel": "Grade 6",
              "interactionTypeCode": "MS",
              "interactionTypeLabel": "Multi Select",
              "isPerformanceItem": true,
              "itemKey": 3615,
              "subjectCode": "MATH",
              "subjectLabel": "Math",
              "targetDescription": "target C description",
              "targetHash": 0,
              "targetId": "C",
              "targetShortName": "",
              "title": "Math Grade 6 Claim 4",
            },
          ],
        ]
      }
      columns={
        Array [
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "item",
              },
            ],
            "compare": [Function],
            "header": "Item",
            "headerClassName": "item",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "stimulus",
              },
            ],
            "compare": [Function],
            "header": "Stimulus ID",
            "headerClassName": "stimulus",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "item-position-in-test",
              },
            ],
            "compare": [Function],
            "header": "Item position in test",
            "headerClassName": "item-position-in-test",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "subject",
              },
            ],
            "compare": [Function],
            "header": "Subject",
            "headerClassName": "subject",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "grade",
              },
            ],
            "compare": [Function],
            "header": "Grade",
            "headerClassName": "grade",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "testname",
              },
            ],
            "compare": [Function],
            "header": "Test name",
            "headerClassName": "testname",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "claim",
                "helpText": [Function],
              },
            ],
            "compare": [Function],
            "header": "Claim",
            "headerClassName": "claim",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "target",
                "helpText": [Function],
              },
            ],
            "compare": [Function],
            "header": "Target",
            "headerClassName": "Target",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "standard",
              },
              Object {
                "accessor": [Function],
                "className": "standard",
                "helpText": [Function],
              },
            ],
            "compare": [Function],
            "header": "Standard",
            "headerClassName": "standard",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "dok",
              },
            ],
            "compare": [Function],
            "header": "DOK",
            "headerClassName": "dok",
          },
          Object {
            "cols": Array [
              Object {
                "accessor": [Function],
                "className": "difficulty",
              },
            ],
            "compare": [Function],
            "header": "Difficulty",
            "headerClassName": "difficulty",
          },
        ]
      }
      index={1}
      isInterimSite={true}
      isLinkTable={false}
      itemIdToTestNameMap={
        Object {
          "2928": Object {
            "testName": "Test 2",
            "testOrder": 1,
          },
          "3163": Object {
            "testName": "Test 1",
            "testOrder": 2,
          },
          "3206": Object {
            "testName": "Test 1",
            "testOrder": 1,
          },
          "3615": Object {
            "testName": "Test 1",
            "testOrder": 3,
          },
        }
      }
      itemSequence={2}
      onAddOrRemoveSelectedItems={[MockFunction]}
      onItemsReorder={[MockFunction]}
      testCodeToLabelMap={
        Object {
          "Test-Name-1": "Test 1",
          "Test-Name-2": "Test 2",
        }
      }
    >
      <tr
        className=""
        key="3163"
      >
        <td>
          <button
            aria-label="Remove selected item from print cart"
            className="btn btn-sm btn-danger btn-remove-item-print-cart"
            onClick={[Function]}
          >
            X
          </button>
        </td>
        <td>
          -
        </td>
        <td>
          3163
        </td>
        <td>
          58679
        </td>
        <td>
          ELA
        </td>
        <td>
          3
        </td>
        <td>
          Test 1
        </td>
        <td>
          2
        </td>
        <td>
          1
        </td>
        <td>
          <ToolTip
            className="box"
            displayIcon={true}
            displayText="B"
            helpText={
              <span>
                Target B description
              </span>
            }
          >
            <span
              className="tool-tip-links box"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <span
                className="tool-tip-hoverable"
              >
                B
                 
                <span
                  aria-hidden="true"
                  className="fa fa-info-circle fa-sm"
                />
              </span>
              <span
                className="tool-tip-details"
              >
                <span
                  className="tool-tip-message  "
                >
                  <span>
                    Target B description
                  </span>
                </span>
              </span>
            </span>
          </ToolTip>
        </td>
        <td>
          <span>
            3.R.2a
             
          </span>
          <span>
            <ToolTip
              className="box"
              displayIcon={true}
              displayText=""
              helpText={
                <span>
                  Content Standard information is currently unavailable for this item.
                </span>
              }
            >
              <span
                className="tool-tip-links box"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                tabIndex={0}
              >
                <span
                  className="tool-tip-hoverable"
                >
                   
                  <span
                    aria-hidden="true"
                    className="fa fa-info-circle fa-sm"
                  />
                </span>
                <span
                  className="tool-tip-details"
                >
                  <span
                    className="tool-tip-message  "
                  >
                    <span>
                      Content Standard information is currently unavailable for this item.
                    </span>
                  </span>
                </span>
              </span>
            </ToolTip>
          </span>
        </td>
        <td />
        <td>
          Easy
        </td>
        <td>
          <div
            className="btn-group"
          >
            <button
              aria-disabled={false}
              aria-label="Move item up"
              className="btn btn-sm btn-primary "
              id="move-item-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-arrow-up"
              />
            </button>
            <button
              aria-disabled={false}
              aria-label="Move item down"
              className="btn btn-sm btn-primary "
              id="move-item-btn"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fa fa-arrow-down"
              />
            </button>
          </div>
        </td>
      </tr>
      <tr
        className="table-row-associated-item"
        key="3163"
      >
        <td>
          <ToolTip
            className="box"
            displayIcon={true}
            displayText=""
            helpText={
              <span>
                This is a Performance Task and must be selected as a group in a predefined sequence. PTs are designed as a complete activity to measure a student’s ability to demonstrate critical-thinking, problem-solving skills and/or complex analysis, and writing and research skills.
              </span>
            }
            position="top"
          >
            <span
              className="tool-tip-links box"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <span
                className="tool-tip-hoverable"
              >
                 
                <span
                  aria-hidden="true"
                  className="fa fa-info-circle fa-sm"
                />
              </span>
              <span
                className="tool-tip-details"
              >
                <span
                  className="tool-tip-message top "
                >
                  <span>
                    This is a Performance Task and must be selected as a group in a predefined sequence. PTs are designed as a complete activity to measure a student’s ability to demonstrate critical-thinking, problem-solving skills and/or complex analysis, and writing and research skills.
                  </span>
                </span>
              </span>
            </span>
          </ToolTip>
        </td>
        <td>
          2
        </td>
        <td>
          3163
        </td>
        <td />
        <td>
          ELA
        </td>
        <td>
          3
        </td>
        <td>
          Test 1
        </td>
        <td>
          2
        </td>
        <td>
          Reading
        </td>
        <td>
          <ToolTip
            className="box"
            displayIcon={true}
            displayText="B"
            helpText={
              <span>
                Target B description
              </span>
            }
          >
            <span
              className="tool-tip-links box"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <span
                className="tool-tip-hoverable"
              >
                B
                 
                <span
                  aria-hidden="true"
                  className="fa fa-info-circle fa-sm"
                />
              </span>
              <span
                className="tool-tip-details"
              >
                <span
                  className="tool-tip-message  "
                >
                  <span>
                    Target B description
                  </span>
                </span>
              </span>
            </span>
          </ToolTip>
        </td>
        <td>
          <span>
            Not Available
             
          </span>
          <span>
            <ToolTip
              className="box"
              displayIcon={true}
              displayText=""
              helpText={
                <span>
                  Content Standard information is currently unavailable for this item.
                </span>
              }
            >
              <span
                className="tool-tip-links box"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                tabIndex={0}
              >
                <span
                  className="tool-tip-hoverable"
                >
                   
                  <span
                    aria-hidden="true"
                    className="fa fa-info-circle fa-sm"
                  />
                </span>
                <span
                  className="tool-tip-details"
                >
                  <span
                    className="tool-tip-message  "
                  >
                    <span>
                      Content Standard information is currently unavailable for this item.
                    </span>
                  </span>
                </span>
              </span>
            </ToolTip>
          </span>
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr
        className="table-row-associated-item"
        key="3206"
      >
        <td>
          <ToolTip
            className="box"
            displayIcon={true}
            displayText=""
            helpText={
              <span>
                This is a Performance Task and must be selected as a group in a predefined sequence. PTs are designed as a complete activity to measure a student’s ability to demonstrate critical-thinking, problem-solving skills and/or complex analysis, and writing and research skills.
              </span>
            }
            position="top"
          >
            <span
              className="tool-tip-links box"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <span
                className="tool-tip-hoverable"
              >
                 
                <span
                  aria-hidden="true"
                  className="fa fa-info-circle fa-sm"
                />
              </span>
              <span
                className="tool-tip-details"
              >
                <span
                  className="tool-tip-message top "
                >
                  <span>
                    This is a Performance Task and must be selected as a group in a predefined sequence. PTs are designed as a complete activity to measure a student’s ability to demonstrate critical-thinking, problem-solving skills and/or complex analysis, and writing and research skills.
                  </span>
                </span>
              </span>
            </span>
          </ToolTip>
        </td>
        <td>
          3
        </td>
        <td>
          3206
        </td>
        <td />
        <td>
          Math
        </td>
        <td>
          6
        </td>
        <td>
          Test 1
        </td>
        <td>
          1
        </td>
        <td>
          Concepts and Procedures and Logical
        </td>
        <td>
          <ToolTip
            className="box"
            displayIcon={true}
            displayText="A"
            helpText={
              <span>
                Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia
              </span>
            }
          >
            <span
              className="tool-tip-links box"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <span
                className="tool-tip-hoverable"
              >
                A
                 
                <span
                  aria-hidden="true"
                  className="fa fa-info-circle fa-sm"
                />
              </span>
              <span
                className="tool-tip-details"
              >
                <span
                  className="tool-tip-message  "
                >
                  <span>
                    Target A description with extra text for testing fewoiauighruiaebiurbbvnianiuergruia
                  </span>
                </span>
              </span>
            </span>
          </ToolTip>
        </td>
        <td>
          <span>
            Not Available
             
          </span>
          <span>
            <ToolTip
              className="box"
              displayIcon={true}
              displayText=""
              helpText={
                <span>
                  Content Standard information is currently unavailable for this item.
                </span>
              }
            >
              <span
                className="tool-tip-links box"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                tabIndex={0}
              >
                <span
                  className="tool-tip-hoverable"
                >
                   
                  <span
                    aria-hidden="true"
                    className="fa fa-info-circle fa-sm"
                  />
                </span>
                <span
                  className="tool-tip-details"
                >
                  <span
                    className="tool-tip-message  "
                  >
                    <span>
                      Content Standard information is currently unavailable for this item.
                    </span>
                  </span>
                </span>
              </span>
            </ToolTip>
          </span>
        </td>
        <td />
        <td />
        <td />
      </tr>
      <tr
        className="table-row-associated-item"
        key="3615"
      >
        <td>
          <ToolTip
            className="box"
            displayIcon={true}
            displayText=""
            helpText={
              <span>
                This is a Performance Task and must be selected as a group in a predefined sequence. PTs are designed as a complete activity to measure a student’s ability to demonstrate critical-thinking, problem-solving skills and/or complex analysis, and writing and research skills.
              </span>
            }
            position="top"
          >
            <span
              className="tool-tip-links box"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <span
                className="tool-tip-hoverable"
              >
                 
                <span
                  aria-hidden="true"
                  className="fa fa-info-circle fa-sm"
                />
              </span>
              <span
                className="tool-tip-details"
              >
                <span
                  className="tool-tip-message top "
                >
                  <span>
                    This is a Performance Task and must be selected as a group in a predefined sequence. PTs are designed as a complete activity to measure a student’s ability to demonstrate critical-thinking, problem-solving skills and/or complex analysis, and writing and research skills.
                  </span>
                </span>
              </span>
            </span>
          </ToolTip>
        </td>
        <td>
          4
        </td>
        <td>
          3615
        </td>
        <td />
        <td>
          Math
        </td>
        <td>
          6
        </td>
        <td>
          Test 1
        </td>
        <td>
          3
        </td>
        <td>
          Modeling/Data Analysis
        </td>
        <td>
          <ToolTip
            className="box"
            displayIcon={true}
            displayText="C"
            helpText={
              <span>
                target C description
              </span>
            }
          >
            <span
              className="tool-tip-links box"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              tabIndex={0}
            >
              <span
                className="tool-tip-hoverable"
              >
                C
                 
                <span
                  aria-hidden="true"
                  className="fa fa-info-circle fa-sm"
                />
              </span>
              <span
                className="tool-tip-details"
              >
                <span
                  className="tool-tip-message  "
                >
                  <span>
                    target C description
                  </span>
                </span>
              </span>
            </span>
          </ToolTip>
        </td>
        <td>
          <span>
            Math Practice
             
          </span>
          <span>
            <ToolTip
              className="box"
              displayIcon={true}
              displayText=""
              helpText={
                <span>
                  Items in this claim primarily measure the Standards for Mathematical Practice rather than Content Standards.
                </span>
              }
            >
              <span
                className="tool-tip-links box"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                tabIndex={0}
              >
                <span
                  className="tool-tip-hoverable"
                >
                   
                  <span
                    aria-hidden="true"
                    className="fa fa-info-circle fa-sm"
                  />
                </span>
                <span
                  className="tool-tip-details"
                >
                  <span
                    className="tool-tip-message  "
                  >
                    <span>
                      Items in this claim primarily measure the Standards for Mathematical Practice rather than Content Standards.
                    </span>
                  </span>
                </span>
              </span>
            </ToolTip>
          </span>
        </td>
        <td />
        <td />
        <td />
      </tr>
    </PrintCartItemTableRow>
  </tbody>
</table>
`;
